### PR TITLE
Prevents NPE on mergeExistingResourcePluginConfiguration when property definition doesn't exist anymore.

### PR DESCRIPTION
**Pull request by rubenvp8510**
@josejulio please review.

### CODE REVIEW

File: modules/core/plugin-container/src/main/java/org/rhq/core/pc/inventory/InventoryManager.java [hunk 1]
```diff
@@ -1945,7 +1945,8 @@ private Configuration mergeExistingResourcePluginConfiguration(Resource resource
             Property discoveredProp = pluginConfig.get(propertyName);
             // Only overrides the property if it is a read-only property
             // See BZ: 1379834
-            if (propDefs.get(propertyName).isReadOnly()) {
+            PropertyDefinition propDefinition = propDefs.get(propertyName);
```
Comment (josejulio) [inline modules/core/plugin-container/src/main/java/org/rhq/core/pc/inventory/InventoryManager.java]: Maybe a bit of paranoia, but, is `propDefs` always non `null` ?
https://github.com/rhq-project/rhq/pull/273/files#diff-32f6b984f4c7846e751c48497839e07aR1938

Comment (burmanm) [inline modules/core/plugin-container/src/main/java/org/rhq/core/pc/inventory/InventoryManager.java]: It's never null. 

https://github.com/rhq-project/rhq/blob/master/modules/core/domain/src/main/java/org/rhq/core/domain/configuration/definition/ConfigurationDefinition.java#L182

Comment (josejulio) [inline modules/core/plugin-container/src/main/java/org/rhq/core/pc/inventory/InventoryManager.java]: Thanks.

**Comment by josejulio** [comment]
LGTM

**Pull request merged**